File: C++/13_Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        int dst_num = 0;
        //unordered_map<char, int> MAP={{'I', 1}, {'V', 5}, {'X', 10}, {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}};
        map<char, int> MAP;
        MAP['I'] = 1;
        MAP['V'] = 5;
        MAP['X'] = 10;
        MAP['L'] = 50;
        MAP['C'] = 100;
        MAP['D'] = 500;
        MAP['M'] = 1000;
        
        for(int i = 0; i < s.size(); i++){
            if(i == 0 || MAP[s[i]] <= MAP[s[i - 1]])
                dst_num += MAP[s[i]];
            else
                dst_num += MAP[s[i]] - 2 * MAP[s[i - 1]];
        }
        return dst_num;
    }
};

File: Python/13_Roman to Integer.py
class Solution:
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        #定义字典
        dst_num = 0
        num_dict = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        for i in range(len(s) - 1):
            if(num_dict[s[i]] < num_dict[s[i + 1]]):
                dst_num -= num_dict[s[i]]
            else:
                dst_num += num_dict[s[i]]
        #把最后一个数加上
        return (dst_num + num_dict[s[-1]])